File: web-nodejs/extension-common/src/model/FlexMediaSection.ts
import {FlexMediaSectionItem} from "./FlexMediaSectionItem";

/**
 * 通用的包含多媒体条目的板块，带一个标题或一些媒体条目信息
 *
 * @author why
 * @since 5/8/24
 */
export interface FlexMediaSection {
    /**
     * 板块标题
     */
    id: string

    /**
     * 板块名称
     */
    title: string

    /**
     * 改板块包含的条目数据
     */
    items: FlexMediaSectionItem[]

    /**
     * 扩展信息
     */
    extras: Map<string, string> | undefined
}

File: lib-extension/src/main/java/com/xiaoyv/flexiflix/extension/model/FlexMediaSection.kt
package com.xiaoyv.flexiflix.extension.model

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

/**
 * [FlexMediaSection]
 *
 * @author why
 * @since 5/8/24
 */
@Parcelize
data class FlexMediaSection(
    @SerializedName("id") var id: String,
    @SerializedName("title") var title: String,
    @SerializedName("items") var items: List<FlexMediaSectionItem>,
    @SerializedName("extras") var extras: HashMap<String, String>? = hashMapOf()
) : Parcelable





